custom_components/wibeee/sensor.py

```diff
@@ -162,9 +167,12 @@ async def fetching_data(now=None):
     return async_track_time_interval(hass, fetching_data, scan_interval)
 
 
-async def async_setup_local_push(hass: HomeAssistant, device, sensors: list['WibeeeSensor']):
+async def async_setup_local_push(hass: HomeAssistant, entry: ConfigEntry, device, sensors: list['WibeeeSensor']):
     mac_address = device['macAddr']
-    nest_proxy = await get_nest_proxy(hass)
+    nest_proxy = await get_nest_proxy(
```

Comment:
Ok so as I mentioned in the other comment this approach is not going to work. But you can see below we have a call and that would be the place to add additional information specific to this device (MAC address).

You could update it to work like this:

```python
nest_proxy.register_device(mac_address, upstream=entry.get(...), listener=on_pushed_data)
```

The `nest` module will then need some refactoring to forward to the configured upstream for that MAC address. Does that make sense? (but don't start just yet as I would like to explore using another approach, which is to have a dropdown selection menu for the upstream)

Reply:
It makes sense that the configuration is for device and not global, just in case there are wibeee devices that have different upstreams, would be strange I guess but possible.

Thank you very much for your help.

It's the first time I try to develop with home assistant, but I'm willing to learn and maybe in the future be able to create my own plugins or addons.

Reply:
Though I'm against the dropdown since there might be tons of upstreams, I saw in a comment somewhere that you mentioned a few upstreams, but non of those where the upstream that I need: my provider is a Spanish company called 'SolarProfit', and they use the endpoint 'http://wdata.solarprofit.es' and the port 8080, that is why I also made the upstream port configurable. What we could make is give a dropdown of options but with the chance of adding your own somehow.

I'll try to fix the change mentioned before and we can further discuss about how to approach the UI.

Reply:
Now I see that making the config by device is not what you meant :upside_down_face: 
Forget about what I said there.

Reply:
Configuration must be per device simply because that is how Home Assistant config flow works. I don't think there's any option (outside of YAML) to make "global" configurations that apply to multiple config entries. When you click "Configure" on an integration those options are for one integration only (one Wibeee device in this case).

https://developers.home-assistant.io/docs/config_entries_config_flow_handler
https://developers.home-assistant.io/docs/config_entries_options_flow_handler

I'm happy to be proven wrong though.

Reply:
Yes, sorry I was confused because the lines appear listed as devices, I'm still trying to figure things out of how home assistant works.

I was trying to use the 'show_advanced_options' but I'm not sure how does it work and I can't find it in the documentation.

Also I found out how to use the dropdown, I just need to know where to find a full list of servers, I just found this:
'http://nest-ingest.wibeee.com',
'http://wdata.solarprofit.es:8080'
'http://wibeee.smilics.com'
'http://monitorconsumo.iberdrola.es'
'http://wattius.mirubee.com'
'http://www.smilics.com:8080'

And those might not exist anymore or have changed, that is why I think we should leave the option to add it written, but maybe instead of having to fields for URL and port it is better to combine into just one field.

Reply:
> I was trying to use the 'show_advanced_options' but I'm not sure how does it work and I can't find it in the documentation.

It probably has something to do with the [per-user setting where you can specify that you are "advanced"](https://www.home-assistant.io/blog/2019/07/17/release-96/#advanced-mode). In which case I would probably not use it.

Could you not do the following in the options flow?
1. Keep the existing checkbox on the first step.
2. When that is submitted then look at the submitted value, if disabled then call `self.async_create_entry` as before but if it is enabled then show a new step with the extra configs that you are adding.

Reply:
I wish it would appear when checking the checkbox, but not sure if that is possible, couldn't find a way to do it in the [voluptuous documentation](https://alecthomas.github.io/voluptuous/docs/_build/html/voluptuous.html)

IMHO having to save the form before getting more options might confuse the users.

Regarding the dropdown, do you still think I should add a dropdown? If so, which upstreams should I add?

Reply:
I made it as radio buttons, I was following and example that was showing up as a drop-down, not sure what I did wrong, but it looks good also with the radio buttons, I left it with the URL but I could also add some text more user friendly like:
![Screenshot from 2023-04-26 12-50-28](https://user-images.githubusercontent.com/6824576/234554616-83295471-4b45-44e7-be4e-8602dc5db623.png)

Reply:
Thanks. Is the PR ready for me to take a look and test?

Do you need the local port to be configurable as in this screenshot?

Reply:
Hi, sorry for the laste answer. Is not that I need it configurable, I just thought would be nice in case some other application/plugin/component already is using that same port.